ui/pages/_app.js

```diff
@@ -363,39 +363,55 @@ class MesheryApp extends App {
   };
 
   loadOrg = async () => {
-    const { store } = this.props;
-    const currentOrgId = sessionStorage.getItem('currentOrgId');
+    const currentOrg = sessionStorage.getItem('currentOrg');
+    let reFetchKeys = false;
+
+    if (currentOrg && currentOrg !== 'undefined') {
+      let org = JSON.parse(currentOrg);
+      await this.loadAbility(org.id, reFetchKeys);
+      this.setOrganization(org);
+    }
+
     dataFetch(
       '/api/identity/orgs',
       {
         method: 'GET',
         credentials: 'include',
       },
-      (result) => {
+      async (result) => {
         let organizationToSet;
 
-        if (currentOrgId !== null) {
-          const indx = result.organizations.findIndex((org) => org.id === currentOrgId);
-          organizationToSet = indx !== -1 ? result.organizations[indx] : result.organizations[0];
+        if (currentOrg) {
+          const indx = result.organizations.findIndex((org) => org.id === currentOrg.id);
+          if (indx === -1) {
+            organizationToSet = result.organizations[0];
+            reFetchKeys = true;
+            await this.loadAbility(organizationToSet.id, reFetchKeys);
+          }
         } else {
           organizationToSet = result.organizations[0];
+          reFetchKeys = true;
+          await this.loadAbility(organizationToSet.id, reFetchKeys);
         }
-
-        store.dispatch({
-          type: actionTypes.SET_ORGANIZATION,
-          organization: organizationToSet,
-        });
-
-        this.loadAbility(organizationToSet.id);
+        this.setOrganization(organizationToSet);
       },
       (err) => console.log('There was an error fetching available orgs:', err),
     );
   };
 
-  loadAbility = async (orgID) => {
+  setOrganization = (org) => {
+    const { store } = this.props;
+    store.dispatch({
+      type: actionTypes.SET_ORGANIZATION,
+      organization: org,
+    });
+  };
+
+  loadAbility = async (orgID, reFetchKeys) => {
```

Comment:
Is this storing org ID and the keys in a cookie?

Reply:
No, it only updates the CASL ability based on keys in sessionStorage if not then it will fetch keys and then updates the ability.